### PR TITLE
Fix #1008: HSET is replacing HMSET 

### LINKED ISSUE #1008: HSET is replacing HMSET and behave in the same way redis 5.0 (reopened from Issue#551)
Hi. I have recently encountered that HSET is not behaving the same way as HMSET. I have found a "stale" issue 551 which was closed. I would like to reopen the discussion and would like to attempt to fix it.

### PR DESCRIPTION
**Pull request by mskuybeda**
I have added functionality to ioredis to support HSET the same way as HMSET. Please Review

### CODE REVIEW

**Comment by mskuybeda** [comment]
About checks not passed, I am not sure why, but issues rise in the files I did not touch. Could you please check what is going on

File: lib/command.ts [hunk 1]
```diff
@@ -397,6 +397,20 @@ Command.setReplyTransformer("hgetall", function(result) {
   return result;
 });
 
+Command.setArgumentTransformer("hset", function(args) {
+  console.log(args.length, args);
```
Comment (luin) [inline lib/command.ts]: Please remove the logs.

File: test/functional/transformer.ts [hunk 2]
```diff
@@ -161,5 +161,44 @@ describe("transformer", function() {
...
+        map.set("b", "e");
+        redis.hset("foo", map, function(err, result) {
+          expect(result).to.eql(2);
+          redis.hget("foo", "b", function(err, result) {
+            expect(result).to.eql("e");
+            done();
+          });
+        });
+      });
+      it("should affect the old way", function(done) {
```
Comment (luin) [inline test/functional/transformer.ts]: I guess you mean "should not" here.

**Comment by luin** [comment]
Thank you for the contribution. The tests are failed. Could you fix it please?

**Comment by mskuybeda** [comment]
Failed tests are in master and not in the files I changed. I will take a look into what causes them, but I am not sure I will have time to fix them this week at least.  Probably will file another issue. I forgot to remove logs, my bad. I will work on the stated changes a bit later today. Thanks for your reply.

**Comment by luin** [comment]
@mskuybeda You're right. I did a quick look and created a fix. Please rebase the latest commit on the master branch.

File: test/functional/transformer.ts (hunk 2 above)
Comment (mskuybeda) [reply test/functional/transformer.ts]: Well, the whole point was to make hset act the same way as hmset. According to [documentation](https://redis.io/commands/hset).

**Comment by luin** [comment]
Cherry-picked to the master branch. Thank you for the contribution!

**Pull request closed without merging**